SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriver.h

```diff
@@ -407,4 +413,10 @@ class SPATIALGDK_API USpatialNetDriver : public UIpNetDriver
 
 	TMultiMap<Worker_EntityId_Key, EActorMigrationResult> MigrationFailureLogStore;
 	uint64 MigrationTimestamp;
+
+	virtual void ServerReplicateActors_BuildConsiderList(TArray<FNetworkObjectInfo*>& OutConsiderList, const float ServerTickTime) override;
+
+	// Store the last received state of the actor over the network by Entity ID so that we can check for non-auth changes.
+	UPROPERTY()
```

Comment:
Could this, and the additional methods, be moved to its own utility class/struct?
Then you can forward declare it and put it in a TUniquePtr like other utility objects. Then you can also just create it if the appropriate setting is set, and just check the pointer to know if you need to perform the associated logic.

Reply:
Good point, thanks. I created a new utility class SpatialNetDriverAuthorityDebugger.